urllib3/util/url.py

```diff
@@ -192,7 +192,10 @@ def parse_url(url):
             # If given, ports must be integers.
             if not port.isdigit():
                 raise LocationParseError(url)
-            port = int(port)
+            try:
```

Comment:
Swapping to regexes here doesn't really work because we allow both unicode and bytes URIs, so we'd need two branches with two different regexes. Given that, it might be better just to drop a comment here that explains the thinking.